src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -106,11 +107,16 @@ public static bool LoadSyntaxTree(Stream stream, out QsCompilation compilation,
             {
                 throw new ArgumentNullException(nameof(stream));
             }
+
+            PerformanceTracking.TaskStart(PerformanceTracking.Task.DeserializerInit);
             using var reader = new BsonDataReader(stream);
+            PerformanceTracking.TaskEnd(PerformanceTracking.Task.DeserializerInit);
             (compilation, reader.ReadRootValueAsArray) = (null, false);
             try
             {
+                PerformanceTracking.TaskStart(PerformanceTracking.Task.SyntaxTreeDeserialization);
                 compilation = Json.Serializer.Deserialize<QsCompilation>(reader);
+                PerformanceTracking.TaskEnd(PerformanceTracking.Task.SyntaxTreeDeserialization);
```

Comment:
This is a great point. Since the exception is not currently caught, it will make the compilation fail at the task it is performing, which will create a misleading error.

I will change it such that it catches the exception, temporarily stores it and generates an appropriate error at the end of compilation when performance data is published.

Thanks Sarah, good catch!

Reply:
Made the changes. Now if something goes wrong while tracking performance, an error is logged at the time results would be published.

Reply:
Sorry, I think my wording wasn't clear. :) I was asking about what if `Json.Serializer.Deserialize` throws an exception? The exception is caught, but `TaskEnd` will never be called. Does that mean the deserialization task will never end?

Reply:
Yes, `TaskEnd` will never be called. When that happens, the error will happen when perf results are published. When the exception is thrown, we catch it in `build.cs` and we log the appropriate error.

Reply:
Sorry, I might be missing it but I don't see where `PublishResults` throws an exception if a task did not end?

Reply:
Sorry, I should have included a bit more details in my comment.

If `TaskEnd` is never called, when calling `PublishResults`, the logic will eventually try to get the duration of the task and since the task did not end, it will throw an `InvalidOperationException` exception with the message "Attempt to get task 'OverallCompilation.ReferenceLoading' duration when no interval has been measured".

The stack at the moment the exception occurs is the following:
```
Microsoft.Quantum.QsCompiler.CommandLineCompiler.CompilationTracker.CompilationTask.get_DurationInMs()
Microsoft.Quantum.QsCompiler.CommandLineCompiler.CompilationTracker.CompilationTaskNode.WriteToJson(Utf8JsonWriter jsonWriter, String prefix)
Microsoft.Quantum.QsCompiler.CommandLineCompiler.CompilationTracker.CompilationTaskNode.WriteToJson(Utf8JsonWriter jsonWriter, String prefix)
Microsoft.Quantum.QsCompiler.CommandLineCompiler.CompilationTracker.PublishResults(String outputFolder)
Microsoft.Quantum.QsCompiler.CommandLineCompiler.BuildCompilation.PublishPerformanceTrackingData(String perfFolder, ConsoleLogger logger)
```

The exception will be caught by `PublishPerformanceTrackingData`, an error will be logged which will also be shown to the user in the console.

Reply:
Thanks for the explanation, that makes sense.